srt-tokio/src/builder.rs

```diff
@@ -208,19 +218,28 @@ impl SrtSocketBuilder {
                 pending_connection::listen(&mut socket, self.init_settings).await?
             }
             ConnInitMethod::Connect(addr, sid) => {
-                pending_connection::connect(
+                let local_addr = self.local_addr.unwrap_or_else(|| unspecified(addr.ip()));
+                println!("{:?} {:?}", addr, local_addr);
```

Comment:
Remove this as well. I'd add a `assert!(addr.is_ipv4() == local_addr.is_ipv4())`, to avoid issues with mismatch.

Reply:
probably it should be an explicit error.

Reply:
I think a panic is reasonable, it's a programming error, I think it's fair to just say it's invalid use of the API